rfcs/rfc-entities-frontend-modernization.md

```diff
@@ -0,0 +1,151 @@
+# RFC: Entities Frontend Modernization (Nuxt 3 + Connect Layer)
+
+**Author:** Cameron Bowler  
+**Status:** DRAFT  
+**Created:** October 1, 2025  
+**Last Updated:** October 3, 2025  
+**Reviewers:** Shahriar Khan, Severin Beauvais, Mikaela Bertucci, Melissa Stanton, Sal Hamood, Thor Wolpert, Dietrich Worlpert  
+
+---
+
+## 1. Purpose
+
+Our Entity UIs currently run on an aging **Vue 2** stack with legacy dependencies (including SBC Common Components). Modernizing to **Nuxt 3** with the **Connect Layer** will reduce technical debt, standardize patterns, and unlock faster change velocity while maintaining feature parity. 
+
+---
+
+## 2. Background
+
+- Several Entity projects are built with **Vue 2 / Vuetify 2**, and rely on SBC Common Components for authentication and other shared UI (e.g., footer, Pay). 
+- A PoC by the **Assets** team successfully upgraded to **Vue 3 / Nuxt** with a Connect extension (still on Vuetify), demonstrating feasibility. 
+- A **direct Nuxt migration** avoids a two-step rewrite (Vue 2 → Vue 3 → Nuxt). 
+
+---
+
+## 3. Proposal
+
+### 3.1 What we’re doing
+- **Pilot on Filings UI** to validate integration patterns and de-risk the broader migration. 
+- Start with a **single-developer foundation for the first 10–15 pts** (framework/bootstrap + core shared upgrades) to avoid conflicts and set consistent patterns; then scale to a small team working in parallel. 
+- Adopt **Nuxt 3 + Connect Layer** as the core, with component strategy (Vuetify 3 -> NuxtUI/Tailwind) aiming for long-term simplification. 
+
+### 3.2 Value
+- Lower maintenance cost, unified authentication via Connect, improved DX, and reuse of shared components—accelerating Create/Edit migrations after the pilot. 
+
+### 3.3 Scope (UIs & filings)
+**Filings UI**: AGM Extension; AGM Location Change; Amalgamation Out; Annual Report; Consent Amalgamation Out; Consent Continuation Out; Continuation Out; Court Order; Notice Of Withdrawal; Director Change Filing (COD); Office Address Change Filing (COA); Non‑filing: Amalgamation Selection; Non‑filing: Digital Business Card pages. 
+
+**Create UI**: Amalgamation; Continuation In; Voluntary Dissolution; Voluntary Dissolution – Firms; Incorporation Application; Registration; Restoration. 
+
+**Edit UI**: Coop Correction; Corp Correction; Firm Correction; Alteration (Company Information); Change of Registration; Firm Conversion; Limited Restoration Extension; Limited Restoration Conversion to Full; Special Resolution. 
+
+---
+
+## 4. Technical Details
+
+### 4.1 Migration Targets
+- Vue 2 → **Vue 3**; Vuetify 2 → **Vuetify 3** *or* **NuxtUI/Tailwind**; **Nuxt 3+** as the application framework. 
+
+### 4.2 Integration Considerations
+- Integrate **Connect Layer** and decouple from SBC Common Components; assess compatibility and regression risks up front. 
+
+### 4.3 Repository Strategy (under evaluation)
```

Comment:
How will this question get resolved? Should it be an option in this RFC like the framework decision?